### PR TITLE
[ldapjs] add client properties describing connected server

### PR DESCRIPTION
**Pull request by tgerk**
client attributes added with PR658, support for multiple servers in connect URL

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: [ldapjs PR#658](https://github.com/ldapjs/node-ldapjs/pull/658/commits) [specific commit](https://github.com/ldapjs/node-ldapjs/pull/658/commits/23e44b2959f18d91ca2bc683a19735ec4878367a), see after line 800
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ldapjs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ldapjs@latest/))
was missing the following properties:
1. Client
1. Server
1. persistentSearch
1. PersistentSearchCache
1. filters

<details>
<summary>as well as these 2 other properties...</summary>
<p>url, parseURL</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6600a9df05523ffcbb86b6f99ddbcef9ad691d19
</p>

File: types/ldapjs/ldapjs-tests.ts [hunk 1]
```diff
@@ -12,6 +12,13 @@ let clientWithMultipleURLs = ldap.createClient({
     ]
 });
 
+// @ts-expect-error
+client.port === 1389;
+client.port === '1389';
+client.host === '127.0.0.1';
+// $ExpectType boolean
+clientWithMultipleURLs.secure;
```
Comment (pmoleri) [inline types/ldapjs/ldapjs-tests.ts:16]: Suggestion. Just for the sake of correctness:
```
client.on('connect', socket => {
    // @ts-expect-error
    client.port === 1389;
    client.port === '1389';
    client.host === '127.0.0.1';
});
clientWithMultipleURLs.on('connect', socket => {
    // $ExpectType boolean
    clientWithMultipleURLs.secure;
});
```

**Review by pmoleri — APPROVED** [review]
Looks good. I added a minor suggestion but type-wise is irrelevant, so you may mark it as resolved if you prefer.

File: types/ldapjs/ldapjs-tests.ts (hunk 1 above)
Comment (tgerk) [reply types/ldapjs/ldapjs-tests.ts]: excellent suggestion!

**Comment by tgerk** [comment]
pending CI, after applying reviewer's suggestion

**Review by pmoleri — APPROVED** [review]

**Comment by tgerk** [comment]
Ready to merge

**Pull request merged**